### ISSUE TITLE
[Enhancement] Gitflow: Rember my starting branch

### ISSUE THREAD

**Issue by JeReT** [ForkIssues/TrackerWin#1717]
**Problem**
We are using git-flow and also use "feature branches" for features that should be merged into a release/*  or hotfix/* branch. The problem is: we always need to manually select "hotfix/1.33.7" from the list, otherwise develop is used. This is super easy to miss.

**Solution A**
if the currently checked out branch is not the selected branch, display a warning that you are starting from a branch that is not the currently active one. This would hopefully a good enough reminder to change the branch accordingly.
Having a button "Start from current branch" which automatically changes the branch in the dropdown would be a very welcome addition.

**Solution B**
Create radio buttons for where to start the branch from:

- develop
- current branch
- custom

If custom, the branch dropdown appears / becomes interactable.
the selection should be remembered for the next time.

**Solution C**
Allow custom branch presets, so the user can define, what is in the branch-dropdown and how the branches are handled. This would also solve requests like [this](https://github.com/fork-dev/TrackerWin/issues/647).
It should be configurable, which is the default starting branch - with a "current branch" wild card.
Not sure if custom commands could be reused / extended here (this could be cool, so we could add other options as well similar to "git flow finish")

**Comment by DanPristupov** [comment]
Hi. GitFlow is a set of [deprecated] rules and also a [deprecated] library. Git-flow made sense a decade ago, when SVN was everywhere and people didn't know how to use git, but now it's not actual. I personally don't understand why 'create a feature' is better than 'create a branch'. If you need to have a 'feature/' or 'release/' prefix, just type them and autocomplete in Fork will help you.

Anyway, gitflow is considered excessive and obsolete even by it's creator and, I'll be honest, most likely we won't improve it.


I'd recommend to try the Create Branch dialog (Ctrl+Shift+B).

<img width="500" alt="Screenshot 2022-12-02 at 14 32 14" src="https://user-images.githubusercontent.com/618115/205304102-89d8305c-67df-4c4d-8517-df24beb0fd01.png">


> **Solution C**
> Allow custom branch presets, so the user can define, what is in the branch-dropdown and how the branches are handled. 

Very soon we will introduce custom commands with UI fields.

**Comment by JeReT** [comment]
the automation of finishing a release- or hotfix-branch which also creates a tag with the version is valuable for us. We retrieve the git tag in a build script to sneak in the version number.

Anyway, Looking forward to see the custom commands in UI fields. That should hopefully solve my problems (as long as there is a "current branch" variable).

**Comment by DanPristupov** [comment]
> as long as there is a "current branch" variable

No, I think there won't be such a variable, but you can request the active branch using `git branch --show-current`
